### PR TITLE
Add recyclerview for homework and used only fragments.

### PR DESCRIPTION
**Pull request by AliaksandrG**
Add recyclerview, fragments, coffee activity
изображение

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeAddFragment.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_coffee_add, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val sort = input_sort_coffee.text
+        val price = input_price_coffee.text
```
Comment (stasenkots) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeAddFragment.kt:27]: this values can be situated in clickListener

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return inflater.inflate(R.layout.fragment_coffee_list, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // RecyclerView node initialized here
+        coffeeListRecyclerView.apply {
+
+            // set a LinearLayoutManager to handle Android
```
Comment (stasenkots) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt:37]: Remove comments

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        coffeePrice?.text = coffeeData.price.toString()
+        Picasso.get().load(coffeeData.URL).error(R.mipmap.ic_error).into(coffeeImage)
+    }
+}
+
+class CoffeeListAdapter(private val list: List<CoffeeData>) :
+    RecyclerView.Adapter<CoffeeViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CoffeeViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
```
Comment (stasenkots) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt:35]: OnCreateViewHolder method  usually  creates the view by inflating an XML layout file and pass it to ViewHolder


  override fun onCreateViewHolder(parent: ViewGroup,
                                    viewType: Int): MyAdapter.MyViewHolder {
        // create a new view
        val textView = LayoutInflater.from(parent.context)
                .inflate(R.layout.my_text_view, parent, false) as TextView
        // set the view's size, margins, paddings and layout parameters
        ...
        return MyViewHolder(textView)
    }


See  https://developer.android.com/guide/topics/ui/layout/recyclerview

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myapplication.R
+import com.squareup.picasso.Picasso
+
+class CoffeeViewHolder(inflater: LayoutInflater, parent: ViewGroup) :
+    RecyclerView.ViewHolder(inflater.inflate(R.layout.list_item_view_coffee, parent, false)) {
+    private var coffeeSort: TextView? = null
```
Comment (stasenkots) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt:13]: I think you can use here lateinit not a nullable type

**Review by stasenkots — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt (hunk 2 above)
Comment (AliaksandrG) [reply app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt:37]: fixed

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeAddFragment.kt (hunk 1 above)
Comment (AliaksandrG) [reply app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeAddFragment.kt:27]: fixed

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt (hunk 4 above)
Comment (AliaksandrG) [reply app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt:13]: fixed

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt (hunk 3 above)
Comment (AliaksandrG) [reply app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt:29]: fixed

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        coffeeListRecyclerView.apply {
+
+            layoutManager = LinearLayoutManager(activity)
+
+            adapter = CoffeeListAdapter(coffeeArray)
+        }
+    }
+
+    companion object {
+        fun newInstance(): CoffeeListFragment = CoffeeListFragment()
```
Comment (TMSlebedev) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeListFragment.kt:33]: what is it for?

File: app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myapplication.R
+import com.squareup.picasso.Picasso
+
+class CoffeeViewHolder(
+    linearView:LinearLayout
```
Comment (TMSlebedev) [inline app/src/main/java/com/example/myapplication/coffeeFragment/CoffeeViewHolder.kt:13]: no need to pass layout to holder

**Review by TMSlebedev — APPROVED** [review]

**Pull request merged**